### DISCUSSION TITLE
Using refactoring rules in a ruleset

### DISCUSSION THREAD

**Discussion by olivernybroe** [JosefPihrt/Roslynator#1089] [General]
**Product and Version Used**: Roslynator.Analyzers, Roslynator.Formatting.Analyzers,: 4.3.0

**Steps to Reproduce**:
Add a small program like so 
```csharp

// My comment
Console.WriteLine("Hello, World!");
```

Add `RR0086 | Remove all comments` rule to ruleset
RR0086	[Remove all comments](https://github.com/JosefPihrt/Roslynator/blob/main/docs/refactorings/RR0086.md)

```xml
<?xml version="1.0" encoding="utf-8" ?>
<RuleSet Name="Custom" ToolsVersion="15.0">
    <Rules AnalyzerId="Roslynator.CSharp.Analyzers" RuleNamespace="Roslynator.CSharp.Analyzers">
        <Rule Id="RR0086" Action="Error"/>
    </Rules>
</RuleSet>
```

Run format command

**Actual Behavior**:
```csharp
// My comment
Console.WriteLine("Hello, World!");
```

**Expected Behavior**:
```csharp
Console.WriteLine("Hello, World!");
```

### Additional information
Thank you for the project! Sorry if this might just be my ignorance, but I am trying to multiple of the refactoring rules to automatically apply when running the formatting command, however it seems like I cannot get it to work. 

When using the analyzer rules it works perfectly. Am I doing something wrong or is it just not supported to do refactoring rules in the ruleset?

**Comment by olivernybroe** [comment]
Oh whoops... Did not see discussions were enabled, feel free to convert it to a discussion instead as it's prob. more appropriate there.

**Comment by josefpihrt** [comment]
Hi,

Unlike analyzers the refactorings cannot be configured through ruleset. Generally speaking, they cannot be configured at all.

Luckily, it is possible to enable/disable Roslynator refactorings using EditorConfig files.

Please see https://github.com/JosefPihrt/Roslynator/blob/main/docs/Configuration.md for further details.

**Reply by olivernybroe** [reply]
Ah alright 👀 Can they still be run automatically on the whole codebase? (not a common scenario, I know)

**Comment by josefpihrt** [comment]
Unfortunately not, refactoring can be considered as a one-time operation invoked at the given location.
